DockerfileLeanFoundation

```diff
@@ -14,35 +14,39 @@ CMD ["/sbin/my_init"]
 
 # Install OS Packages:
 # Misc tools for running Python.NET and IB inside a headless container.
-RUN \
-  apt-get update && \
-  apt-get install -y wget xvfb unzip curl libxrender1 libxtst6 libxi6 python-pip && \
-  pip install --upgrade pip && pip install --upgrade cython && \
-  pip install pycparser pandas scipy numpy sklearn  && \
-  pip install blaze cvxopt cvxpy pykalman statistics statsmodels arch copulalib && \
-  pip install xgboost theano keras tensorflow deap
+RUN apt-get update && apt-get install -y wget bzip2 python3-pip python-opengl && \
+    apt-get install -y xvfb unzip curl libxrender1 libxtst6 libxi6
+
+# Install miniconda and supported third party python packages
+ENV PATH="/opt/miniconda3/bin:${PATH}"
+RUN wget https://repo.continuum.io/miniconda/Miniconda3-4.3.31-Linux-x86_64.sh && \
+    bash Miniconda3-4.3.31-Linux-x86_64.sh -b -p /opt/miniconda3 && \
+    rm -rf Miniconda3-latest-Linux-x86_64.sh && \
```

Comment:
I think we should use:
```
RUN wget https://repo.continuum.io/miniconda/Miniconda3-4.3.31-Linux-x86_64.sh -O miniconda.sh && \
    bash miniconda.sh -b -p /opt/miniconda3 && \
    rm -rf miniconda.sh && \
```
To keep the version in only one line. In this case we were deleting a non-existing file.

Reply:
I actually did something similar you shared with me, but when I ran it, docker couldn't find the renamed script, so I dd it this way instead.